action/protocol/rolldpos/epoch.go

```diff
@@ -64,8 +67,43 @@ func (p *Protocol) Handle(context.Context, action.Action, protocol.StateManager)
 }
 
 // ReadState read the state on blockchain via protocol
-func (p *Protocol) ReadState(context.Context, protocol.StateManager, []byte, ...[]byte) ([]byte, error) {
-	return nil, protocol.ErrUnimplemented
+func (p *Protocol) ReadState(ctx context.Context, sm protocol.StateManager, method []byte, args ...[]byte) ([]byte, error) {
+	switch string(method) {
+	case "NumCandidateDelegates":
+		return byteutil.Uint64ToBytes(p.numCandidateDelegates), nil
+	case "NumDelegates":
+		return byteutil.Uint64ToBytes(p.numDelegates), nil
+	case "NumSubEpochs":
+		return byteutil.Uint64ToBytes(p.numSubEpochs), nil
+	case "NumSubEpochsDardanelles":
```

Comment:
this looks odd. We changed the number of delegates before, but no method to read the previous value. Shall we just have "NumSubEpochs"?